Domain/Entry.php

```diff
@@ -31,17 +31,7 @@ class Entry implements AuditableEntryInterface
     private $auditSuccess;
     private $granting;
 
-    /**
-     * Constructor.
-     *
-     * @param int    $id
-     * @param string $strategy
-     * @param int    $mask
-     * @param bool   $granting
-     * @param bool   $auditFailure
-     * @param bool   $auditSuccess
-     */
-    public function __construct($id, AclInterface $acl, SecurityIdentityInterface $sid, $strategy, $mask, $granting, $auditFailure, $auditSuccess)
+    public function __construct(?int $id, AclInterface $acl, SecurityIdentityInterface $sid, string $strategy, int $mask, bool $granting, bool $auditFailure, bool $auditSuccess)
```

Comment:
why allowing `null` as id ?

Reply:
Seems like this package is missing some `|null` unions in the PHPdocs.

We sometimes set the id to null: https://github.com/symfony/security-acl/blob/eb1ddd923910a32635dedb30ececf338a29ba54b/Domain/Acl.php#L481 

And we also take care of handling null in "some" places: https://github.com/symfony/security-acl/blob/eb1ddd923910a32635dedb30ececf338a29ba54b/Dbal/MutableAclProvider.php#L841